src/internal/endpoint-runtime.ts

```diff
@@ -0,0 +1,143 @@
+import type { JsonValue } from 'type-fest'
+import uuid from 'tiny-uid'
+import { serializeError } from 'serialize-error'
+import type { DataTypeKey, Destination, GetDataType, GetReturnType, IBridgeMessage, IInternalMessage, OnMessageCallback, RuntimeContext } from '../types'
+import { parseEndpoint } from './endpoint'
+
+export interface EndpointRuntime {
+  sendMessage: <ReturnType extends JsonValue, K extends DataTypeKey = DataTypeKey>(
+    messageID: K,
+    data: GetDataType<K, JsonValue>,
```

Comment:
This is currently a required param even if `ProtocolMap` is not extended. This means that the user won't be able to do this:
```ts
sendMessage('FOO') // oops, we require a second type param
```

Was this intended?

Reply:
my personal nit: this wouldn't be quite an issue at all if the API would require this:
```ts
sendMessage({ type: 'FOO' })
```

Reply:
FWIW (probably not much), I have definitely found myself often writing `sendMessage('FOO', {})` because of the _required_ second argument